### PR TITLE
MM-27688 - Implement OpenId Connect in Cloud

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://mattermost.atlassian.net/browse/MM-27688

Updated:
- Cloud Admin Guide > Authentication > Office 365 Single Sign-On
   - Added H2s to break up steps & supplemental images
   - Updated procedure for creating a new connection via OpenID Connect

### CODE REVIEW

File: source/cloud/cloud-administration/sso-office.md [hunk 1]
```diff
@@ -3,36 +3,75 @@ ___
 
 Follow these steps to configure Mattermost to use your Office 365 logon credentials and Azure Active Directory account as a Single Sign-on (SSO) service for team creation, account creation, and sign-in.
 
-**The system must be using SSL for use with Office365 as Microsoft only allows OAuth redirect URIs that are SSL enabled.**
+**Note:** The system must be using SSL for use with Office365 as Microsoft only allows OAuth redirect URIs that are SSL enabled.
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-office.md:6]: @sbishel Given that SSL is considered older, less secure terminology (deprecated in favour of TLS), does it make sense to update this note to use TLS instead? CC @wiersgallak

File: source/cloud/cloud-administration/sso-google.md [hunk 2]
```diff
@@ -2,49 +2,77 @@
...
-8. Restart your Mattermost server to see the changes take effect.
+4. Restart your Mattermost server to see the changes take effect.
 
 **Note:**
-- You may also enter **Client ID** and **Client Secret** fields from step 6 directly in `GoogleSettings` section of `config/config.json`.
-- The following default values in `GoogleSettings` section of `config/config.json` are recommended:
+- Alternatively, you may also enter **Client ID** and **Client Secret** values directly into the `GoogleSettings` section of the Mattermost `config/config.json` file.
+- The following default values are recommended:
     - `Scope`: `profile email`
     - `AuthEndpoint`: `https://accounts.google.com/o/oauth2/v2/auth`
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-google.md:77]: @sbishel - Anything here that needs to change with regards to supporting OpenID Connect over OAuth 2?

File: source/cloud/cloud-administration/sso-google.md [hunk 3]
```diff
@@ -2,49 +2,77 @@
...
     - `TokenEndpoint`: `https://www.googleapis.com/oauth2/v4/token`
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-google.md:78]: @sbishel - Anything here that needs to change with regards to supporting OpenID Connect over OAuth 2?

File: source/cloud/cloud-administration/sso-office.md [hunk 4]
```diff
@@ -3,36 +3,75 @@ ___
...
+
+4. Paste in the _Application (client) ID_ as the **Client ID**
+
+5. Paste in the _Directory (tenant) ID_ as the **Directory (tenant) ID**.
+
+6. Open the Mattermost `config.json` file.
+
+7. In the `Office365Settings` section of `config.json` file, update the following default values:
+
+   - ``AuthEndpoint: https://login.microsoftonline.com/common/oauth2/v2.0/authorize``
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-office.md:66]: @sbishel - Anything here that needs to change with regards to supporting OpenID Connect over OAuth 2?

File: source/cloud/cloud-administration/sso-office.md [hunk 5]
```diff
@@ -3,36 +3,75 @@ ___
...
+   - ``TokenEndpoint: https://login.microsoftonline.com/common/oauth2/v2.0/token``
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-office.md:67]: @sbishel - Anything here that needs to change with regards to supporting OpenID Connect over OAuth 2?

File: source/cloud/cloud-administration/sso-office.md [hunk 6]
```diff
@@ -3,36 +3,75 @@ ___
...
+
+to include the _Directory (tenant) ID_ value, for example:
+
+   - ``AuthEndpoint: https://login.microsoftonline.com/3dfccc41-d360-488a-a6ff-e9e565b3xbf1/oauth2/v2.0/authorize``
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-office.md:71]: @sbishel - Anything here that needs to change with regards to supporting OpenID Connect over OAuth 2?

File: source/cloud/cloud-administration/sso-office.md [hunk 7]
```diff
@@ -3,36 +3,75 @@ ___
...
+   - ``TokenEndpoint: https://login.microsoftonline.com/3dfcc41-d360-488a-a6ff-e9e565b3xbf1/oauth2/v2.0/token``
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-office.md:72]: @sbishel - Anything here that needs to change with regards to supporting OpenID Connect over OAuth 2?

File: source/cloud/cloud-administration/sso-gitlab.rst [hunk 8]
```diff
@@ -1,15 +1,28 @@
...
 =====================
 
-Configure Mattermost to use GitLab as a Single Sign-on (SSO) service for team creation, account creation, and user sign-in.
+Follow these steps to configure Mattermost to use GitLab as a Single Sign-on (SSO) service for team creation, account creation, and user sign-in.
 
-.. Note:: Only the default GitLab SSO is officially supported. "Double SSO", where GitLab SSO is chained to other SSO solutions, is not supported. It may be possible to connect GitLab SSO with AD, LDAP, SAML, or MFA add-ons in some cases, but because of the special logic required they're not officially supported and are known not to work on some experiences. 
+.. Note::  
+  Only the default GitLab SSO is officially supported. "Double SSO", where GitLab SSO is chained to other SSO solutions, is not supported. It may be possible to connect GitLab SSO with AD, LDAP, SAML, or MFA add-ons in some cases, but because of the special logic required, they're not officially supported, and they're known not to work in some cases. 
 
+Migrating an existing connection
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/sso-gitlab.rst:9]: @sbishel - Where can I see the migration workflow in action (for customers that already have a connection set up?)

**Comment by cwarnermm** [comment]
@wiersgallak - We have product documentation for setting up Office 365, Google, and GitLab with OpenID Connect. Does it make sense to add another page that details how to setting up a generic OpenID application?

**Comment by wiersgallak** [comment]
@cwarnermm Yes, I think a page for generic OpenID connect makes sense. Thanks!

File: source/cloud/cloud-administration/sso-gitlab.rst [hunk 9]
```diff
@@ -21,20 +34,31 @@ Step 1: Add an OAuth application to your GitLab account
...
+4. Copy and paste the *Application Secret Key* from GitLab to Mattermost. 
+5. Enter the **Gitlab Site URL** of our GitLab instance. 
+
+  If your GitLab instance is not set up to use SSL, the endpoints must begin with ``http://`` instead of ``https://``. If you are using GitLab itself as your OAuth provider, use *gitlab.com*.
+
+6. Enter the **User API Endpoint**. 
+
+  Use ``https://{gitlab-site-name}/api/v3/user`` if you're running GitLab v8.17.8 or earlier.
+
+(Optional) Step 3: Forcing all users to sign up with SSO only
```
Comment (wiersgallak) [inline source/cloud/cloud-administration/sso-gitlab.rst:59]: Should this be "Force all users to sign up with SSO only" seems like the verb tense changes for this step.

File: source/cloud/cloud-administration/sso-google.md [hunk 10]
```diff
@@ -2,49 +2,77 @@
...
-![](../../../source/images/google_3_oauth_client_id.png)
+4. Select **Credentials** and select **Create credentials**, then choose **OAuth client ID** from the drop-down list.
 
 5. Select **Web Application** as the **Application type**, and choose a descriptive **Name** for the OAuth connection.
 
-Under **Restrictions > Authorized redirect URIs**, enter `your-mattermost-url/signup/google/complete` (example: `http://localhost:8065/signup/google/complete`). Click **Create**.
+6. Under **Restrictions > Authorized redirect URIs**, enter `your-mattermost-url/signup/google/complete` (example: `http://localhost:8065/signup/google/complete`), then select **Create**.
 
-![](../../../source/images/google_4_web_app.png)
+7. Copy and paste the **Client ID** and **Client Secret** values to a temporary location. You will enter these values in the Mattermost System Console.
```
Comment (sbishel) [inline source/cloud/cloud-administration/sso-google.md:45]: These steps have changed. Here are the steps as defined in the UI.
Screen Shot 2021-02-09 at 2 17 35 PM

I think we will probably need new screen shots as well.

File: source/cloud/cloud-administration/sso-google.md (hunk 2 above)
Comment (sbishel) [reply source/cloud/cloud-administration/sso-google.md:77]: Yes. I would simply remove "AuthEndpoint", "TokenEndpoint", "UserApiEndpoint"
```
    "GoogleSettings": {
        "Enable": false,
        "Secret": "fake_secret",
        "Id": "fake_id",
        "Scope": "profile openid email",
        "AuthEndpoint": "",
        "TokenEndpoint": "",
        "UserApiEndpoint": "",
        "DiscoveryEndpoint": "https://accounts.google.com/.well-known/openid-configuration",
        "ButtonText": "",
        "ButtonColor": ""
    },
    ```

File: source/cloud/cloud-administration/sso-google.md (hunk 3 above)
Comment (sbishel) [reply source/cloud/cloud-administration/sso-google.md:78]: see above.

File: source/cloud/cloud-administration/sso-office.md (hunk 1 above)
Comment (sbishel) [reply source/cloud/cloud-administration/sso-office.md]: Have we changed it in other places? SSL is generally used in a more generic sense and is probably ok to leave here.

File: source/cloud/cloud-administration/sso-office.md [hunk 11]
```diff
@@ -3,36 +3,79 @@ ___
...
+
+3. Paste in the _Directory (tenant) ID_ from Azure Portal as the **Directory (tenant) ID** in Mattermost.
+
+4. The **Discovery Endpoint** for OpenID Connect with Office 365 is prepopulated with ``https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration``.
+
+5. Paste in the _Application (client) ID_ from Azure Portal as the **Client ID** in Mattermost.
+
+6. Paste in the _client secret value_ from Azure Portal as the **Client Secret** in Mattermost.
+
+6. Open the Mattermost `config.json` file.
```
Comment (sbishel) [inline source/cloud/cloud-administration/sso-office.md:56]: This would be for manual update. I don't think steps 6 and 7 belong here.

**Review by sbishel — APPROVED** [review]

File: source/cloud/cloud-administration/sso-google.md (hunk 2 above)
Comment (cwarnermm) [reply source/cloud/cloud-administration/sso-google.md:77]: @wiersgallak - We're currently inconsistent about providing both System Console-based instructions and config.json-based instructions for System Admins to configure service providers. We can either improve our instructions around what goes directly into config.json, or we can remove the config-json instructions from the topics that include them. What's your preference? Do we know if System Admins find the config.json file-based instructions useful?

File: source/cloud/cloud-administration/sso-office.md (hunk 1 above)
Comment (cwarnermm) [reply source/cloud/cloud-administration/sso-office.md]: Fair point - we do use the term "SSL" throughout the docs to refer to something more generic. We'll leave this as-is. Thanks!

File: source/cloud/cloud-administration/sso-office.md (hunk 11 above)
Comment (cwarnermm) [reply source/cloud/cloud-administration/sso-office.md:56]: @sbishel - I was confused by this too. This is pre-existing content in the docs, and while it feels like the manual way to edit directly via the config.json file, it also read like a couple of additional manual steps that are required specifically for Office 365. Can you confirm either way, please? If everything beyond the "open the MM config.json file" step is no longer needed for Office 365, I'll happily take it out!

File: source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst [hunk 12]
```diff
@@ -0,0 +1,26 @@
+Converting OAuth 2.0 to OpenID Connect
+=======================================
+
+OAuth 2.0 is being deprecated and replaced by OpenID Connect. You need to migrate your OAuth configuration to the new OpenID Connect standard. 
+[within what timeframe?] [failure to do so will result in....tbd]
```
Comment (cwarnermm) [inline source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst:5]: @wiersgallak @sbishel Can you provide some details about the urgency around converting, please? Is there a deadline that should be adhered to? Are there consequences to System Admins not converting ASAP?

File: source/cloud/cloud-administration/sso-gitlab.rst [hunk 13]
```diff
@@ -1,40 +1,56 @@
 GitLab Single Sign-On
 =====================
 
-Configure Mattermost to use GitLab as a Single Sign-on (SSO) service for team creation, account creation, and user sign-in.
+Migrating from OAuth 2.0 to OpenID Connect
+-------------------------------------------
+
+OAuth 2.0 is being deprecated and replaced by OpenID Connect. Refer to product documentation to `convert your existing OAuth configuration <https://docs.mattermost.com/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect>`__ for GitLab to the new OpenID Connect standard.
```
Comment (sbishel) [inline source/cloud/cloud-administration/sso-gitlab.rst:7]: This is ok for cloud. But I just want to make sure for the on-premise edition, the TE will continue to allow gitlab via OAUTH.

File: source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst [hunk 14]
```diff
@@ -0,0 +1,26 @@
+Converting OAuth 2.0 to OpenID Connect
+=======================================
+
+OAuth 2.0 is being deprecated and replaced by OpenID Connect. You need to migrate your OAuth configuration to the new OpenID Connect standard.
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst:4]: Removed "new" from the 3 supported service provider topics and the conversion topic.

File: source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst (hunk 12 above)
Comment (cwarnermm) [reply source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst:5]: Thanks! I've adjusted the tone of the sentence to feel less urgent/time sensitive.

File: source/cloud/cloud-administration/sso-office.md (hunk 11 above)
Comment (sbishel) [reply source/cloud/cloud-administration/sso-office.md]: Yep, re-reviewed and confirmed the two steps should be removed. This is no longer necessary. I looked at the current documentation and it appears out of date.

**Comment by cwarnermm** [comment]
@wiersgallak @justinegeffen The Cloud-specific product documentation updates for OpenID Connect are now ready for final PM and Editor review. Thanks for your patience on this one!

File: source/cloud/cloud-administration/sso-google.md (hunk 2 above)
Comment (wiersgallak) [reply source/cloud/cloud-administration/sso-google.md]: @cwarnermm Many customers use the config file directly, so I think its best if we improve them.  Agree that improving them can come at a later time though.

**Review by wiersgallak — APPROVED** [review]
Thanks @cwarnermm looks good!

**Review by justinegeffen — COMMENTED** [review]
Just a few comments from me - but nothing blocking. Looks awesome, well done. :)

File: source/cloud/cloud-administration/sso-google.md [hunk 15]
```diff
@@ -1,51 +1,78 @@
...
+
+1. Go to [Google Cloud Platform](https://console.developers.google.com).
+
+2. Select **Credentials** in the left-hand sidebar.
+
+3. Select **Create Credentials**, then select **OAuth client ID**.
+
+4. select the **Web application** as the application type.
+
+5. Enter `Mattermost - your-company-name` as the **Name**,
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply source/cloud/cloud-administration/sso-google.md:23]: @justinegeffen They're entering their own company name following "Mattermost - "

File: source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst [hunk 16]
```diff
@@ -0,0 +1,26 @@
+
+Converting OAuth 2.0 to OpenID Connect
+=======================================
+
+OAuth 2.0 is being deprecated in the System Console and replaced by OpenID Connect. Using the System Console, you can migrate your OAuth configuration to the OpenID Connect standard.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst:5]:
```suggestion
OAuth 2.0 is being deprecated and replaced by OpenID Connect. Using the System Console, you can migrate your OAuth configuration to the OpenID Connect standard. 
```

File: source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst [hunk 17]
```diff
@@ -0,0 +1,26 @@
+
+Converting OAuth 2.0 to OpenID Connect
+=======================================
+
+OAuth 2.0 is being deprecated in the System Console and replaced by OpenID Connect. Using the System Console, you can migrate your OAuth configuration to the OpenID Connect standard. 
+
+.. image:: ../../images/OAuth20Deprecated.png
+
+A one-time, one-click conversion tool is available within the existing **OAuth 2.0** page or within the new **OpenID Connect** page by going to **System Console > Authentication**.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst:9]:
```suggestion
A one-time, one-click conversion tool is available within the existing **OAuth 2.0** page. You can also go to **System Console > Authentication > OpenID Connect**.
```

File: source/cloud/cloud-administration/sso-google.md [hunk 18]
```diff
@@ -1,51 +1,78 @@
...
+1. Log in to Mattermost, then go to **System Console > Authentication > OpenID Connect**.
 
-Under **Restrictions > Authorized redirect URIs**, enter `your-mattermost-url/signup/google/complete` (example: `http://localhost:8065/signup/google/complete`). Click **Create**.
+2. Select **Google Apps** as the service provider.
 
-![](../../../source/images/google_4_web_app.png)
+3. The **Discovery Endpoint** for OpenID Connect with Google Apps is prepopulated with ``https://accounts.google.com/.well-known/openid-configuration``.
 
-6. Copy the **Client ID** and **Client Secret**, which will be used to configure Google SSO for Mattermost.
+3. Paste in the **Client ID** from Google in Mattermost.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-google.md:53]:
```suggestion
4. Paste in the **Client ID** from Google in Mattermost.
```

File: source/cloud/cloud-administration/sso-google.md [hunk 19]
```diff
@@ -1,51 +1,78 @@
...
 
-![](../../../source/images/google_5_client_id_secret.PNG)
+3. Paste in the **Client Secret** from Google in Mattermost.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-google.md:55]:
```suggestion
5. Paste in the **Client Secret** from Google in Mattermost.
```

File: source/cloud/cloud-administration/sso-google.md [hunk 20]
```diff
@@ -1,51 +1,78 @@
...
 
-7. In **System Console > OAuth 2.0 > Select OAuth 2.0 service provider**, choose **Google Apps** as the service provider and enter **Client ID** and **Client Secret** from step 6 in their respective fields.
+4. Select **Save**.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-google.md:57]:
```suggestion
6. Select **Save**.
```

File: source/cloud/cloud-administration/sso-google.md [hunk 21]
```diff
@@ -1,51 +1,78 @@
...
 
-8. Restart your Mattermost server to see the changes take effect.
+5. Restart your Mattermost server to see the changes take effect.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-google.md:59]:
```suggestion
7. Restart your Mattermost server to see the changes take effect.
```

File: source/cloud/cloud-administration/sso-office.md [hunk 22]
```diff
@@ -1,38 +1,61 @@
...
 
 Follow these steps to configure Mattermost to use your Office 365 logon credentials and Azure Active Directory account as a Single Sign-on (SSO) service for team creation, account creation, and sign-in.
 
-**The system must be using SSL for use with Office365 as Microsoft only allows OAuth redirect URIs that are SSL enabled.**
+**Note:** The system must be using SSL for use with Office365 as Microsoft only allows OAuth redirect URIs that are SSL enabled.
+
+### Step 1: Register an application in Azure Portal
 
-1. Log in to your [Azure Portal](https://portal.azure.com/) with the account that relates to the Azure AD tenant you want to register the application in. You can check the tenant in the top right corner of the portal.
+1. Log in to your [Azure Portal](https://portal.azure.com/) with the account that relates to the Azure Active Directory tenant where you want to register the application. You can confirm the tenant in the top right corner of the portal.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-office.md:15]:
```suggestion
1. Log in to the [Azure Portal](https://portal.azure.com/) with the account that relates to the Azure Active Directory tenant where you want to register the application. You can confirm the tenant in the top right corner of the portal.
```

File: source/cloud/cloud-administration/sso-office.md [hunk 23]
```diff
@@ -1,38 +1,61 @@
...
+
+1. Log in to Mattermost, then go to **System Console > Authentication > OpenID Connect**.
+
+2. Select **Office 365** as the service provider.
+
+3. Paste the Directory (tenant) ID from Azure Portal as the **Directory (tenant) ID** in Mattermost.
+
+4. The **Discovery Endpoint** for OpenID Connect with Office 365 is prepopulated with ``https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration``.
+
+5. Paste the Application (client) ID from Azure Portal as the **Client ID** in Mattermost.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-office.md:52]:
```suggestion
5. Paste the **Application (client) ID** from the Azure Portal as the **Client ID** in Mattermost.
```

File: source/cloud/cloud-administration/sso-gitlab.rst [hunk 24]
```diff
@@ -1,40 +1,57 @@
...
-6. [Optional] To force all users to sign-up with SSO only, set **System Console > Authentication > Email > Enable sign-in with email** to ``false``
-
-Users must change their sign-in method before they can sign in with GitLab.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Log in to Mattermost, then go to **System Console > Authentication > OpenID Connect**.
+2. Select **GitLab** as the service provider.
+3. Enter the **Gitlab Site URL** of your GitLab instance. If your GitLab instance is not set up to use SSL, start the URL with ``http://`` instead of ``https://``. If you are using GitLab itself as your provider, use ``gitlab.com``.
+4. The **Discovery Endpoint** for OpenID Connect with GitLab is prepopulated with ``https://gitlab.com/.well-known/openid-configuration``.
+5. Paste the *Application ID* from GitLab as the **Client ID** in Mattermost.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-gitlab.rst:48]:
```suggestion
5. Paste the **Application ID** from GitLab as the **Client ID** in Mattermost.
```

File: source/cloud/cloud-administration/sso-gitlab.rst [hunk 25]
```diff
@@ -1,40 +1,57 @@
...
+6. Paste the *Application Secret Key* from GitLab as the **Client Secret** in Mattermost.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-gitlab.rst:49]:
```suggestion
6. Paste the **Application Secret Key** from GitLab as the **Client Secret** in Mattermost. 
```

File: source/cloud/cloud-administration/sso-office.md [hunk 26]
```diff
@@ -1,38 +1,61 @@
...
    ![](../../../source/images/AzureApp_App_Directory_IDsv2.png)
 
-6. Log in to Mattermost and then go to the **System Console > OAuth 2.0 > Select OAuth 2.0 service provider**, choose **Office 365** as the service provider. Enter the _client secret value_ you copied in Step 4 as the **Application Secret Password**. Paste the _Application (client) ID_ you saved in Step 5 into the **Application ID** field.
+### Step 3: Configure Mattermost for Office 365 SSO
+
+1. Log in to Mattermost, then go to **System Console > Authentication > OpenID Connect**.
+
+2. Select **Office 365** as the service provider.
+
+3. Paste the Directory (tenant) ID from Azure Portal as the **Directory (tenant) ID** in Mattermost.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-office.md:48]:
```suggestion
3. Paste the **Directory (tenant) ID** from the Azure Portal as the **Directory (tenant) ID** in Mattermost.
```

File: source/cloud/cloud-administration/sso-office.md [hunk 27]
```diff
@@ -1,38 +1,61 @@
...
+
+4. The **Discovery Endpoint** for OpenID Connect with Office 365 is prepopulated with ``https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration``.
+
+5. Paste the Application (client) ID from Azure Portal as the **Client ID** in Mattermost.
+
+6. Paste the client secret value from Azure Portal as the **Client Secret** in Mattermost.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-office.md:54]:
```suggestion
6. Paste the **client secret** value from the Azure Portal as the **Client Secret** in Mattermost.
```

Comment (justinegeffen) [reply source/cloud/cloud-administration/sso-office.md:54]: Confirm whether the three fields I've edited above have the correct casing to match the UI (the other steps have different casing for these fields).

**Review by justinegeffen — COMMENTED** [review]
I re-reviewed this and made some additional comments/suggestions. :)

File: source/cloud/cloud-administration/sso-google.md [hunk 28]
```diff
@@ -1,51 +1,78 @@
...
+
+### Step 1: Create OpenID Connect project in Google API Manager
+
+1. Go to [Google Cloud Platform](https://console.developers.google.com).
+
+2. Select **Credentials** in the left-hand sidebar.
+
+3. Select **Create Credentials**, then select **OAuth client ID**.
+
+4. select the **Web application** as the application type.
```
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-google.md:21]:
```suggestion
4. Select the **Web application** as the application type.
```

File: source/cloud/cloud-administration/sso-google.md (hunk 15 above)
Comment (justinegeffen) [inline source/cloud/cloud-administration/sso-google.md:23]:
```suggestion
5. Enter `Mattermost-<your-company-name>` as the **Name**, replacing <your-company-name> with the name of your organization.
```

File: source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst (hunk 16 above)
Comment (cwarnermm) [reply source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst:5]: @justinegeffen In this specific case, OAuth 2.0 is being deprecated only in the UI (the System Console). System Admins aren't being forced to migrate their existing OAuth 2.0 configurations to OpenID Connect (though we give them a quick way to do so). Given this context, would you continue to recommend that we remove the reference to the System Console here?

File: source/cloud/cloud-administration/sso-google.md (hunk 2 above)
Comment (cwarnermm) [reply source/cloud/cloud-administration/sso-google.md]: Logged https://mattermost.atlassian.net/browse/MM-32932 for this follow-on docs update.

File: source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst (hunk 16 above)
Comment (justinegeffen) [reply source/cloud/cloud-administration/converting-oauth-2.0-to-openid-connect.rst:5]: Thanks for the context, and no, agreed that this would not work.

**Review by justinegeffen — APPROVED** [review]

**Pull request merged**